core/drivers/clk/clk.c

```diff
@@ -285,18 +398,23 @@ TEE_Result clk_set_parent(struct clk *clk, struct clk *parent)
 	size_t pidx = 0;
 	uint32_t exceptions = 0;
 	TEE_Result res = TEE_ERROR_GENERIC;
+	struct clk_elt *clk_elt = NULL;
 
-	if (clk_get_parent_idx(clk, parent, &pidx) || !clk->ops->set_parent)
+	if (!clk->ops->set_parent)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	if (clk_get_parent_idx(clk, parent, &pidx))
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
Not sure why you splat the `if` here, it still works with with lw_clk since clk_get_parent_idx would return `TEE_ERROR_BAD_PARAMETERS`. I don't think it's needed, it only save a call which is done in some error case.

Reply:
agree. get_parent will succeed.